inc/functions.php

```diff
@@ -357,8 +357,8 @@ function my_date($format, $stamp="", $offset="", $ty=1, $adodb=false)
 			$dstcorrection = $mybb->settings['dstcorrection'];
 		}
 
-		// If DST correction is enabled, add an additional hour to the timezone.
-		if($dstcorrection == 1)
+		// If DST correction is enabled and the date falls into DST, add an additional hour to the timezone.
+		if($dstcorrection == 1 || ($dstcorrection == 2 && date('I', $stamp) == 1))
```

Comment:
Probably for the best. 2.0 needs a user setting for the country that would be considered for DST calculations at this point.